client/src/app/templates/new/templates-new.component.ts

```diff
@@ -1,84 +1,116 @@
+        private toasterService: ToasterService,
+        private router: Router
     ) {
-        this.createForm();
+        this.createForms();
     }
 
-    createForm(): void {
-        this.serviceForm = this.fb.group({
-            startFormControl: new FormControl('', [Validators.required]),
-            endFormControl: new FormControl('', [Validators.required]),
-            dayFormControl: new FormControl('', [Validators.required]),
-            nbMaxFormControl: new FormControl('', [Validators.required]),
+    createForms(): void {
+        this.templateNameFormGroup = this.fb.group({
+            templateNameFormControl: ['', Validators.required],
         });
-        this.formArray = this.fb.array([
-            this.serviceForm,
+        this.servicesFormArray = this.fb.array([]);
+        this.generalFormArray = this.fb.array([
+            this.templateNameFormGroup,
+            this.servicesFormArray,
         ]);
-        this.templateForm = this.fb.group({
-            nameFormControl: new FormControl('', [Validators.required]),
-            formArray: this.formArray,
+        this.generalFormGroup = this.fb.group({
+            generalFormArray: this.generalFormArray,
+        });
+    }
+
+    addServiceForm(nbMax: Number, startAt: Date, endAt: Date, startDay: Number, endDay: Number): void {
+        const serviceFormGroup = this.fb.group({
+            startFormControl: [startAt, Validators.required],
+            startDayFormControl: [startDay, Validators.required],
+            endFormControl: [endAt, Validators.required],
+            endDayFormControl: [endDay, Validators.required],
+            nbMaxFormControl: [nbMax, Validators.required],
         });
+        serviceFormGroup.valueChanges.subscribe(() => {
+            this.sortServiceForm();
+        });
+        this.servicesFormArray.push(serviceFormGroup);
+    }
+
+    sortServiceForm(): void {
+        this.servicesFormArray.controls.sort((a, b) => {
+            const aStartAt = (a as FormGroup).controls.startAtFormControl.value;
+            const bStartAt = (b as FormGroup).controls.startAtFormControl.value;
+            if (aStartAt < bStartAt) {
+                return -1;
+            } else if (aStartAt > bStartAt) {
+                return 1;
+            }
+            return 0;
+        });
+    }
+
+    getControls(): Array<AbstractControl> {
+        return (this.generalFormArray.get([1]) as FormArray).controls;
+    }
+
+    removeServiceForm(fromGroupId: Number): void {
+        this.servicesFormArray.removeAt(+fromGroupId);
     }
 
-    toNumber(date: Date, selectedDay): { day: Number, hours: Number, minutes: Number } {
+    toNumber(date: String, selectedDay): { day: Number, hours: Number, minutes: Number } {
         return {
             day: selectedDay,
-            hours: date.getHours(),
-            minutes: date.getMinutes(),
+            hours: Number(date.split(':')[0]),
+            minutes: Number(date.split(':')[1]),
```

Comment:
I saw the use of `+something` to conert to number earlier. We should maybe use the same pattern everywhere, what do you think?

Reply:
Yes